modules/navier_stokes/test/src/kernels/MallocKernel.C

```diff
@@ -0,0 +1,36 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MallocKernel.h"
+#include "SystemBase.h"
+
+registerMooseObject("NavierStokesTestApp", MallocKernel);
+
+InputParameters
+MallocKernel::validParams()
+{
+  return Kernel::validParams();
+}
+
+MallocKernel::MallocKernel(const InputParameters & parameters) : Kernel(parameters) {}
+
+void
+MallocKernel::jacobianSetup()
+{
+  auto & sys_mat = _sys.getMatrix(_sys.systemMatrixTag());
+
+  // Add to an off-diagional
+  sys_mat.add(0, 1, 0);
```

Comment:
I feel like `.add(0, N, 0)` would be better. `N` is the global column size of sys_mat. 

I was worried about if we increase mesh, this may not trigger malloc error anymore.

Reply:
I don't think increasing the mesh size should have any effect here. The only thing that would change this being an error is if we switched the default dof ordering from variable major to node major. Because here I'm using PJFNK with default diagonal coupling. 0 is dof for variable u; 1 is dof for variable v

Reply:
Is it not a different error if you try to add outside the bounds of the matrix? Wouldn't doing an addition at (0, N) actually be outside the bounds? I would think you can only index up to N - 1.

Reply:
OK, I see. I understood now. I missed that you are using PJFNK.